src/main/java/de/retest/recheck/AssertionMessage.java

```diff
@@ -0,0 +1,61 @@
+package de.retest.recheck;
+
+import java.io.File;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import de.retest.recheck.printer.TestReplayResultPrinter;
+import de.retest.recheck.report.ActionReplayResult;
+import de.retest.recheck.report.TestReplayResult;
+import de.retest.recheck.ui.diff.InsertedDeletedElementDifference;
+import de.retest.recheck.ui.diff.LeafDifference;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+class AssertionMessage {
```

Comment:
`AssertionMessage` is really generic and in my opinion does two things. Do we want to have specific error message classes for both `getNoGoldenMasterErrorMessage` and `getDifferencesErrorMessage`?

Reply:
We can use `RecheckAssertionMessage` or `RecheckCapMessage` if you want to?

However, I wouldn't go for separate classes in this case. It already moves complexity out of `RecheckImpl` into a class, which is solely concerned with creating a proper assertion message for the given test result data. Therefore, testing became way easier.

I don't think there is much benefit in further splitting this up into separate types. Either `RecheckImpl` would have to decide which one to create, or you have something like a factory that does this.

Reply:
> We can use `RecheckAssertionMessage` or `RecheckCapMessage` if you want to?

Yes, sure, just to be more precise. Although I would prefer `RecheckCapMessage` to make sure it is not misused in some other place.

> I don't think there is much benefit in further splitting this up into separate types

My general rule of thumb (with few exceptions and respective [builder](https://dzone.com/articles/too-many-parameters-java-1)) is not to have a class/method with more than 3 parameters as indicated by https://rules.sonarsource.com/java/RSPEC-107. Sadly this is not the only class in our code.
 _So maybe a step at a time._

Reply:
I've adapted the class name, definitely better now.

I also agree that the 5 ctor args that are currently used aren't beautiful, but how would splitting up the implementation help here? `getDifferencesErrorMessage()` needs all of them.

Reply:
Well, maybe address this separately.